src/components/customer-resource-coverage/customer-resource-coverage.js

```diff
@@ -226,21 +233,147 @@ class CustomerResourceCoverage extends Component {
   }
 }
 
-const validate = (values, props) => {
-  moment.locale(props.locale);
+/**
+ * Validator to ensure start date comes before end date chronologically
+ * @param {} dateRange - coverage date range to validate
+ * @returns {} - an error object if errors are found, or `false` otherwise
+ */
+const validateStartDateBeforeEndDate = (dateRange) => {
+  const message = 'Start date must be before end date';
+
+  if (dateRange.endCoverage && moment(dateRange.beginCoverage).isAfter(moment(dateRange.endCoverage))) {
+    return { beginCoverage: message };
+  }
+
+  return false;
+};
+
+/**
+ * Validator to ensure begin date is present and entered dates are valid
+ * @param {} dateRange - coverage date range to validate
+ * @returns {} - an error object if errors are found, or `false` otherwise
+ */
+const validateDateFormat = (dateRange, locale) => {
+  moment.locale(locale);
   let dateFormat = moment.localeData()._longDateFormat.L;
-  let errors = [];
+  const message = `Enter date in ${dateFormat} format.`;
 
-  values.customCoverages.forEach((dateRange, index) => {
-    let dateRangeErrors = {};
+  if (!dateRange.beginCoverage || !moment(dateRange.beginCoverage).isValid()) {
+    return { beginCoverage: message };
+  }
+
+  return false;
+};
+
+/**
+ * Validator to ensure all coverage ranges are within the parent package's
+ * custom coverage range if one is present
+ * @param {} dateRange - coverage date range to validate
+ * @param {} packageCoverage - parent package's custom coverage range
+ * @param {} intl - object containing locale-specific data & formatting
+ * @returns {} - an error object if errors are found, or `false` otherwise
+ */
+const validateWithinPackageRange = (dateRange, packageCoverage, intl) => {
+  // javascript/moment has no mechanism for "infinite", so we
+  // use an absurd future date to represent the concept of "present"
```

Comment:
Is this descriptive enough for someone new to code based? Like will they understand that 'present' is Folio terms is an open-ended end-date that is rolling? Is there documentation in Folio where you read up in the concept of 'present' per-say? Just a thought.

Reply:
i think it's descriptive enough for the code it's referencing or, as you put it, for someone new to the codebase.  for someone new to FOLIO in general, probably not.  but i'd argue this isn't the right place to delineate what "present" means in this context - we'd probably want to do that in documentation, in user-facing information prompts or (if done as a code comment) somewhere more general than this validation method.

that's just my feeling on it though - since it's a comment i think more information wouldn't be likely to hurt anything.  if you wanna reword it just let me know - you wrote the line anyway so you should probably get first dibs at describing it :) . looks like i got some linting errors to fix before merging anyway

Reply:
This is fine with me.